lib/url.js

```diff
@@ -144,11 +141,7 @@ function urlParse(url, parseQueryString, slashesDenoteHost) {
 }
 
 function isIpv6Hostname(hostname) {
-  return (
-    StringPrototypeCharCodeAt(hostname, 0) === CHAR_LEFT_SQUARE_BRACKET &&
-    StringPrototypeCharCodeAt(hostname, hostname.length - 1) ===
-    CHAR_RIGHT_SQUARE_BRACKET
-  );
+  return hostname.startsWith('[') && hostname.endsWith(']');
```

Comment:
What about the primordials? Are they not important?

CC @nodejs/primordials

Reply:
How Is using startsWith and endsWith can be considered being faster as it claims that it has even better performance?